### DISCUSSION TITLE
HttpClient dependency injection not supported in future versions of Azure Functions

### DISCUSSION THREAD

**Discussion by marcin0klusek** [Azure/azure-functions-dotnet-worker#2457] [Feedback]
.Net 6.0 and Azure Functions v4

Today I saw the log in AppInsights "Using HttpClient as an injected dependency will not be supported in future versions of Azure Functions. Use IHttpClientFactory instead. See http://aka.ms/functions-httpclient-di for more information." but I can't find any announcement about that.

Is there a post about future versions of Azure Functions? Do you know where I can find more information about unsupported dependency injections?

**Comment by kshyju** [comment]
The documentation you linked is for the .NET6 in-process model function apps. This repo is for the isolated model.  See https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-in-process-differences

.NET Isolated model is vey similar to how you write a .NET core console app. So whatever DI things you can do there in the .net core console app, you will be able to do here in the isolated model.
